[rocprofiler-compute] Update Documentation for Strix Halo (#4794)
ROCm/rocm-systems: projects/rocprofiler-compute/docs/conceptual/rdna/command-processor.rst

.. meta::

>    :description: ROCm Compute Profiler — RDNA3.5 command processor / CPC metrics

Reviewer: ```suggestion
   :description: Learn about the Command Processor Compute (CPC) metrics in ROCm Compute Profiler, including utilization, interface activity, stalls, and cache behavior on RDNA 3.5 (gfx1151).
```

   :keywords: ROCm Compute Profiler, RDNA, gfx1151, command processor, CPC

.. _rdna-command-processor:

> =========================

Reviewer 2: Align format, CDNA uses * for titles and here we use =. This comment applies to all of the RDNA files.

Reviewer: Heading levels are determined based on heading succession. However, it's best practice to follow the standard convention and style format for consistency:
<img width="397" height="207" alt="image" src="https://github.com/user-attachments/assets/e02f479f-53ef-4b52-b663-a1acda294f9e" />

Reviewer 3: will change in the next PR

Command processor (CP)
=========================

The command processor (CP) connects the host and kernel driver to on-GPU

> scheduling: it pulls work from HSA queues, decodes packets, and hands kernel

Reviewer: ```suggestion
scheduling. During the process it pulls work from HSA queues, decodes packets, and dispatches the kernel
```

> launches to the front-end (SPI / WGP path). On Instinct GPUs the profiler

Reviewer: ```suggestion
launches to the front-end (SPI / WGP path). On Instinct GPUs, the profiler
```

> launches to the front-end (SPI / WGP path). On Instinct GPUs the profiler

Reviewer: ```suggestion
launches to the front end (SPI/WGP path). On Instinct GPUs, the profiler
```

> often splits metrics into CPF (fetcher) and CPC (packet processor); the

Reviewer: ```suggestion
often seperates the metrics into CPF (fetcher) and CPC (packet processor). The
```

> shipped gfx1151 analysis panels emphasize CPC and MEC activity (utilization,

Reviewer 4: Can we at least write our explicitly that MEC is micro engine compute? I know we don't have a page for this, because this would have to be another separate ticket for explaining the MEs and scheduler.

> shipped gfx1151 analysis panels emphasize CPC and MEC activity (utilization,

Reviewer: Do we need to explicitly mention "shipped"? If not can be removed or replaced with "The default gfx1151 analysis panels".

> shipped gfx1151 analysis panels emphasize CPC and MEC activity (utilization,

Reviewer: ```suggestion
shipped gfx1151 analysis panels emphasize CPC and MEC activity, including utilization,
```

> interface, stalls, memory requests, instruction cache).

Reviewer: ```suggestion
interface utilization, stall cycles, memory requests, and instruction cache).
```

> often seperates the metrics into CPF (fetcher) and CPC (packet processor). The

Reviewer 2: Not critical in this PR, but we should fix the abbreviation here, where the first occurance should be the full name.
For example:
```suggestion
often seperates the metrics into command processor fetcher (CPF) and command processor packet processor (CPC). The
```

> For the full CDNA narrative and CPF / CPC metric tabs across MI-series arches, see

Reviewer: ```suggestion
For the complete CDNA architecture overview and the CPF and CPC metric tabs across MI-series GPUs, see
```

> shipped gfx1151 analysis panels emphasize CPC and ME (Micro Engine) activity, including utilization,

Reviewer 4: ```suggestion
gfx1151 analysis panels emphasize CPC and ME (Micro Engine) activity, including utilization,
```

> :doc:`../cdna/command-processor`.

Reviewer: ```suggestion
:doc:`../cdna/command-processor` under CDNA-CDNA4.
```

> Command Processor Compute (CPC) — gfx1151

Reviewer 4: Do we want to keep gfx1151 in all of the headers, or preemptively remove for when we add the other gfx11 product support?
This is non-critical ask

> Command Processor Compute (CPC) — gfx1151

Reviewer: ```suggestion
Command processor compute (CPC) — gfx1151
```

> interface utilization, stall cycles, memory requests, and instruction cache).

Reviewer 2: ```suggestion
interface utilization, stall cycles, memory requests, and instruction cache.
```
no matching opening bracket.

===========================================

> 

Reviewer: As @cfallows-amd suggested, this section needs some explanation of MEC before we introduce the MEC Stall Cycles and MEC Instruction Cache.

Reviewer 3: have changed all occurrence of MEC to "ME(Micro Engine)"

Reviewer 3: I have changed all MEC to ME(Micro Engine)

> CPC Utilization

Reviewer: ```suggestion
CPC utilization
```

---------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-cpc-utilization-gfx1151
         :file: _templates/metrics_table.j2

> CPC Interface Utilization

Reviewer: ```suggestion
CPC interface utilization
```

-------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-cpc-interface-utilization-gfx1151
         :file: _templates/metrics_table.j2

> MEC Stall Cycles

Reviewer: ```suggestion
MEC stall cycles
```

----------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-mec-stall-cycles-gfx1151
         :file: _templates/metrics_table.j2

> CPC Memory Requests

Reviewer: ```suggestion
CPC memory requests
```

-------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-cpc-memory-requests-gfx1151
         :file: _templates/metrics_table.j2

> MEC Instruction Cache

Reviewer: ```suggestion
MEC instruction cache
```
